### DISCUSSION TITLE
When should I use a storage class vs a persistent volume?

### DISCUSSION THREAD

**Discussion by yrro** [kubernetes-csi/csi-driver-smb#590] [Q&A] [answered]
I'm currently using persistent volumes that I create manually for my users whenever there's a need to access a share from pods running in a namespace.

Can storage classes make this easier? Or otherwise, when would you use storage classes vs persistent volumes?

**Comment by andyzhangx** [comment, marked as answer]
@yrro with storage class, the driver could create a subDir dynamically, and if you want to create dozens of PVCs(e.g. statefulset), using storage class is the easier way, while if you only want to create one pvc and don't need to create a subDir(e.g. bring your own sub dir), then using pv also works.

**Reply by yrro** [reply]
Thanks :)
